ecm_prep_web.py

```diff
@@ -0,0 +1,7361 @@
+            # Assume secondary effects pertain to all fuel types and
+            # technologies associated with the given secondary end use(s)
+            self.fuel_type, self.technology = [
+                {"primary": x, "secondary": "all"} for x in [self.fuel_type, self.technology]
+            ]
+            # Assume secondary heating/cooling effects pertain to demand-side
+            # heating/cooling technologies (e.g., secondary 'technology_type'
+            # is set to 'demand')
+            if (
+                isinstance(self.end_use["secondary"], list)
+                and any(
+                    [
+                        x not in ["heating", "secondary heating", "cooling"]
+                        for x in self.end_use["secondary"]
+                    ]
+                )
+            ) or (
+                not isinstance(self.end_use["secondary"], list)
+                and self.end_use["secondary"] not in ["heating", "secondary heating", "cooling"]
+            ):
+                self.technology_type = {
+                    "primary": self.technology_type,
+                    "secondary": "supply",
+                }
+            else:
+                self.technology_type = {
+                    "primary": self.technology_type,
+                    "secondary": "demand",
+                }
+        # Case where the user has not flagged any secondary end use impacts;
+        # set 'secondary' key values for relevant attributes to None
+        else:
+            (
+                self.energy_efficiency,
+                self.energy_efficiency_units,
+                self.fuel_type,
+                self.end_use,
+                self.technology,
+                self.technology_type,
+            ) = [
+                {"primary": x, "secondary": None}
+                for x in [
+                    self.energy_efficiency,
+                    self.energy_efficiency_units,
+                    self.fuel_type,
+                    self.end_use,
+                    self.technology,
+                    self.technology_type,
+                ]
+            ]
+
+        # Fill out an 'all' climate zone input
+        if self.climate_zone == "all" or "all" in self.climate_zone:
+            self.climate_zone = self.handyvars.in_all_map["climate_zone"]
+
+        # Fill out an 'all' structure type input
+        if self.structure_type == "all" or "all" in self.structure_type:
+            self.structure_type = self.handyvars.in_all_map["structure_type"]
+
+        # Fill out an 'all' building type, fuel type, end use, and/or
+        # technology input. Note that these attributes are affected by whether
+        # or not the user has flagged secondary end use impacts
+        for mseg_type in [x[0] for x in self.end_use.items() if x[1] is not None]:
+            # Fill out a building type input that is marked 'all',
+            # 'all residential' or 'all commercial', or is formatte
```

Comment:
I don't think we need this function. The node backend will feed ECMs into the ecm table, so we don't need to populate that table from the file system.